src/components/Verse/VerseAndTranslation.tsx

```diff
@@ -32,37 +59,56 @@ interface Props {
 
 const VerseAndTranslation: React.FC<Props> = (props) => {
   // If fixedFontScale is provided as a prop, use it; otherwise, get from hook (Redux)
-  const { fixedFontScale, ...restProps } = props;
+  const { fixedFontScale, chapter, ...restProps } = props;
   const {
     data,
     error,
     mutate,
     translationFontScale: reduxTranslationFontScale,
     quranTextFontScale: reduxQuranTextFontScale,
-  } = useVerseAndTranslation(restProps);
-  const { arabicVerseClassName, translationClassName, quranFont } = props;
+  } = useVerseAndTranslation({ ...restProps, chapter });
+
+  const {
+    data: chapterData,
+    error: chapterError,
+    isLoading: chapterIsLoading,
+    mutate: mutateChapter,
+  } = useChapter({
+    chapterIdOrSlug: chapter.toString(),
+  });
+
+  const handleRetry = useCallback(() => {
+    mutate();
+    mutateChapter();
+  }, [mutate, mutateChapter]);
 
-  if (error) return <Error error={error} onRetryClicked={mutate} />;
+  if (error || chapterError) {
+    return <Error error={error || chapterError} onRetryClicked={handleRetry} />;
+  }
 
-  if (!data) return <Spinner />;
+  if (!data || chapterIsLoading) return <Spinner />;
 
   return (
     <div className={styles.container}>
       {data?.verses?.map((verse) => (
         <div key={verse.verseKey} className={styles.verseContainer}>
-          <div className={classNames(styles.arabicVerseContainer, arabicVerseClassName)}>
+          <div className={classNames(styles.arabicVerseContainer, restProps.arabicVerseClassName)}>
             <PlainVerseText
-              quranFont={quranFont}
+              shouldShowTitle
```

Comment:
[CRITICAL-REGRESSION] this change is going to change other components not in the homepage since this component is shared between multiple components in other pages!


<img width="1040" height="881" alt="Image" src="https://github.com/user-attachments/assets/9ad5152f-4195-42d3-87e8-fa4d1dfb46a4" />